packages/core/src/lib/markdown-it/index.js

```diff
@@ -140,13 +142,32 @@ markdownIt.renderer.rules.fence = (tokens, idx, options, env, slf) => {
   // wrap all lines with <span> so we can number them
   str = lines.map((line, index) => {
     const currentLineNumber = index + 1;
-    const rule = highlightRules.find(highlightRule => highlightRule.shouldApplyHighlight(currentLineNumber));
-    if (rule) {
-      return rule.applyHighlight(line, currentLineNumber);
+    // Rules that affects this line
+    const rules = highlightRules.filter(
+      highlightRule => highlightRule.shouldApplyHighlight(currentLineNumber));
+    if (rules.length === 0) {
+      // not highlighted
+      return `<span>${line}\n</span>`;
+    }
+    const rawBoundaries = [];
+    let highlightType = HIGHLIGHT_TYPES.PartialText;
+    // Per line WholeLine override WholeText overrides PartialText
```

Comment:
Just a confirmation, is the priority existing behaviour too?

Reply:
If so seems like an odd choice in retrospect.. we should probably have done the opposite to allow things like this. 🤔🤔 

```js
// Let something more specific override the more generic one
16-18,16[2:10]
```

But for another time.

Reply:
> For the opposite, do you mean 16[2:10] would mean that the whole line 16 highlight rule will be ignored?

Yes, a better usage example would be:

```
// I want to highlight a bunch of lines easily without breaking up the rules
// but just a specific chunk in line 20
highlight-lines="16-36,20[2:10]"
```

It's very low priority too as there is an easy alternative (breakup `16-36` into 2).

> What about the case 16[1:10], 16[3:5]? Will that mean just keeping 16[3:5]?

Hmm I think this has little use case (cmiiw) since it deals with the finest granularity of character/word bounds on the same line. We can leave it be to implementation details but try to keep the behaviour consistent.

The new implementation here merges disjoint/overlapping bounds on the same line under `collateAllIntervals`, so that could be what we should aim for going forward. `16[1:10], 16[3:5]` merged into `16[1:10]`

Reply:
@zlimez can you update the comment per suggestion pls.

Reply:
Done